compiler/rustc_passes/src/liveness.rs

```diff
@@ -766,81 +781,48 @@ impl<'a, 'tcx> Liveness<'a, 'tcx> {
             return false;
```

Comment:
we already check this at the start of `union` so we may not need to check this here as well.

We also never used `first_merge` here, might make sense to remove that :thinking: Is this helpful when looking at the logs?

Reply:
The motivation for a check is somewhat different. In `RWUTable` it avoids unnecessary work and failing a runtime aliasing check, in `merge_from_succ` it detects that there is de-facto no-merge and avoids logging. I would rather leave it be even if it is redundant.

I don't know about `first_merge`.

Reply:
> it detects that there is de-facto no-merge and avoids logging.

:+1: 

> I don't know about first_merge.

Looking at the code using `merge_from_succ` I am slightly in favor of removing it

Reply:
Ok. I will look into removing `first_merge` in a follow-up PR.